feat: add new `toggleSelected` option

ghiscoding opened the pull request:
- new option accepts a boolean or a callback returning a boolean
- should fix number 6 of Discussion #244
- the idea came from Air Datepicker [toggleSelected](https://air-datepicker.com/docs)

uvarov-frontend commented:
Thank you for your input, I will definitely look at all your changes when I have time.

ghiscoding commented:
> > @ghiscoding Now when I started digging into the code, I discovered that this option was already there before. `settings.selection.cancelableDay` Now I'm thinking what should I do with two identical options. ‭😀
> 
> Ow, well toggleSelected is a little more intuitive. :)

oh wow ok I did not see that when I added the code, I also think `toggleSelected` is more intuitive and the one I added also has the advantage to work both a `boolean` and a callback function. Lastly I purposely chose the same name as the other Air Picker library, who knows we might get users who migrate to your library :)
